libsawtooth/src/client/rest.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Bitwise IO
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A client for interacting with sawtooth services.
+
+use reqwest::blocking::Client;
+use serde::Deserialize;
+use std::collections::VecDeque;
+
+use crate::client::Batch as ClientBatch;
+use crate::client::SawtoothClient;
+
+pub use super::error::SawtoothClientError;
+
+/// A client that can be used to interact with sawtooth services. This client handles
+/// communication with the REST API.
+pub struct RestApiSawtoothClient {
+    url: String,
+}
+
+impl RestApiSawtoothClient {
+    /// Create a new 'RestApiSawtoothClient' with the given base 'url'. The URL should be the bind
+    /// endpoint of the sawtooth REST API.
+    pub fn new(url: &str) -> Self {
+        Self { url: url.into() }
+    }
+}
+
+impl SawtoothClient for RestApiSawtoothClient {
+    /// Get all existing batches in the current blockchain.
```

Comment:
This doc should be on the trait's declaration of the function.

Reply:
Fixed